File: Python/StackWithLinkList/stack.py
from node import Node

class Stack:


    def __init__(self):
        self.top = None


    def isEmpty(self):
        return self.top == None


    def peek(self):
        return self.top.val


    def push(self, node:Node):
        if not self.isEmpty():
            node.down = self.top
            self.top = node

        else:
            self.top = node
        print('Node added')


    def pop(self):
        if not self.isEmpty():
            tmp = self.top.val
            self.top = self.top.down
            print('Node deleted !')
            return tmp
        else:
            print('There is nothing to delete !')


    def printStack(self):
        if not self.isEmpty():
            tmp = self.top
            while tmp is not None:
                print(tmp.val, end=' ')
                tmp = tmp.down
            print()
        else:
            print('There is nothing to show !')

File: C-plus-plus/StackWithArray/Stack.cpp
#include "Node.cpp"
#include <iostream>

using namespace std;

class Stack {
public:
    int top;
    int size;
    Node **arr;

    Stack(int size) {
        this->size = size;
        top = -1;
        arr = new Node*[size];
    }

    bool isEmpty() {
        return top == -1;
    }

    bool isFull() {
        return top == size;
    }

    Node* peek() {
        if (!isEmpty()) {
            return arr[top];
        }
        cout << "Stack is empty !" << endl;
        return nullptr;
    }

    void push(Node *node) {
        if (!isFull()) {
            arr[++top] = node;
        }
        else {
            cout << "Stack is Full" << endl;
        }
        cout << "Node added" << endl;
    }

    Node* pop() {
        if (!isEmpty()) {
            top--;
            return arr[top+1];
        }
        cout << "Stack is empty !" << endl;
        return nullptr;
    }

    void printStack() {
        for (int i = top; i > -1; i--) {
            cout << arr[i]->val << " ";
        }
        cout << endl;
    }
};